config/consensus.go

```diff
@@ -1057,9 +1057,31 @@ func initConsensusProtocols() {
 	// v29 can be upgraded to v30, with an update delay of 7 days ( see calculation above )
 	v29.ApprovedUpgrades[protocol.ConsensusV30] = 140000
 
+	v31 := v30
+	v31.ApprovedUpgrades = map[protocol.ConsensusVersion]uint64{}
+	v31.EnableBatchVerification = true
+	v31.RewardsCalculationFix = true
+	v31.MaxProposedExpiredOnlineAccounts = 32
+
+	// Enable TEAL 6 / AVM 1.1
+	v31.LogicSigVersion = 6
+	v31.EnableInnerTransactionPooling = true
+	v31.IsolateClearState = true
+
+	// stat proof key registration
+	v31.EnableStateProofKeyregCheck = true
+
+	// Maximum validity period for key registration, to prevent generating too many StateProof keys
+	v31.MaxKeyregValidPeriod = 256*(1<<16) - 1
```

Comment:
I was about to ask some questions about the numbers going into this formula, seems like intent is to address this in a follow on issue: #3257 cc @algonautshant

Reply:
The assumption is that the source of the truth is here: `v31.MaxKeyregValidPeriod = 256*(1<<16) - 1`
Anywhere else should be referencing this parameter. 

The issue in #3257 is to make sure this variable is used for testing (instead of hardcoding the same value elsewhere), and fix the package dependency issue.

This is not a blocking issue.

Reply:
`MaxKeyregValidPeriod` combines 256 and 16, which are based on two different limits. 
for example: 
merklearray/merkle.go:	MaxEncodedTreeDepth = 16